### DISCUSSION TITLE
How to use in my fabric mod

### DISCUSSION THREAD

**Discussion by Panda98277** [maruohon/malilib#128] [Mod Development]

**Comment by maruohon** [comment]
Currently there is no written documentation, so the best way to see how to use malilib for your mods is to look at for example Litematica, MiniHUD or Tweakeroo.

For the most part you just need to create a class that implements `IInitializationHandler`, and then there you register your configs, hotkeys, renderers and whatever else you might have into the malilib systems.

So for example in Litematica:
https://github.com/maruohon/litematica/blob/pre-rewrite/fabric/1.19.x/src/main/java/fi/dy/masa/litematica/InitHandler.java

And then you register that init handler in your main mod initializer like so:
https://github.com/maruohon/litematica/blob/4e55dcf2c4923bbd59e85a509683acec1129177b/src/main/java/fi/dy/masa/litematica/Litematica.java#L13-L17

And you can look at those implementations and interfaces to see how they work.

Also note that there is the big malilib rewrite coming at some point. It's getting pretty close to being "usable for the mass audience", but I'm not sure when I'll actually release it for the general player base on CurseForge or Modrinth. So depending on your mod's development timeline, it may or may not make sense to target that new malilib code instead. It simplifies a lot of things and reduces the amount of unnecessary boilerplate code. And it's a lot more flexible in many ways, for example properly supporting custom config types without having to extend and override like half of the malilib config screen stuff like in the current old code... :pain:
